frameworks/Java/servlet/src/main/java/hello/Cache2kPostgresServlet.java

```diff
@@ -29,39 +39,51 @@
 	@Override
 	public void init(ServletConfig config) throws ServletException {
 		super.init(config);
+		randomNumbers = new CircularList<>(shuffledNumbers(DB_ROWS));
 
 		Map<Integer, CachedWorld> worlds;
 		try {
 			worlds = Common.loadAll(dataSource.getConnection());
 		} catch (SQLException e) {
 			throw new ServletException(e);
 		}
-		
+
 		// Build the cache
-		cache = new Cache2kBuilder<Integer, CachedWorld>() {}
-		    .name("cachedWorld")
-		    .eternal(true)
-		    .entryCapacity(DB_ROWS)
-		    .build();
+		cache = new Cache2kBuilder<Integer, CachedWorld>() {
+		}.name("cachedWorld").eternal(true).entryCapacity(DB_ROWS).build();
 		cache.putAll(worlds);
 	}
 
 	@Override
 	protected void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException,
 			IOException {
 		final int count = Common.normalise(req.getParameter("queries"));
-		final Random random = ThreadLocalRandom.current();
+		final int start = ThreadLocalRandom.current().nextInt(DB_ROWS);
+		List<Integer> keys = IntStream.range(start, start + count)
+				.mapToObj(i -> randomNumbers.get(i)).collect(Collectors.toList());
```

Comment:
While `start` will indeed be random on all requests, the order of the queried ids will never change after startup.

This must be changed to loop `count` times and perform `nextInt` and `get` each time through the loop to be merged.

Reply:
@msmith-techempower I've already noted that the current implementation is [not following the rules](https://github.com/TechEmpower/FrameworkBenchmarks/pull/3244#issuecomment-362849176) for the test `cache query` test.
@nbrady-techempower left it open for some additional feedback of my thoughts.